### PR TITLE
Added syntax for the structurizr DSL

### PR DESCRIPTION
**Pull request by venthur**
structurizr DSL is used in structurizr, which is a tool that allows to
model your software architecture according to the c4 model
(https://c4model.com/)

The default file extension is .dsl but it clashes with an existing (but
non-compatible) filetype.

I've added the following test, but I'm not expert in vimscript:

```vim
" dsl
au BufNewFile,BufRead *.dsl
	\ if getline(1) =~ '^<\!' |
	\   setf dsl |
	\ else |
	\   setf structurizr |
	\ endif
```
we test if the `.dsl` file starts with `<!`, in which case it would be a `dsl` file, otherwise it would be `structurizr`

The language reference can be found here:
https://github.com/structurizr/dsl

### CODE REVIEW

File: runtime/filetype.vim [hunk 1]
```diff
@@ -538,7 +538,12 @@ au BufNewFile,BufRead *.drac,*.drc,*lvs,*lpe	setf dracula
 au BufNewFile,BufRead *.ds			setf datascript
 
 " dsl
-au BufNewFile,BufRead *.dsl			setf dsl
+au BufNewFile,BufRead *.dsl
+	\ if getline(1) =~ '^<\!' |
```
Comment (dkearns) [inline runtime/filetype.vim:542]: You might want to use `getline(nextnonblank(1))` here.

**Comment by dkearns** [comment]
Shouldn't the `dsl` filetype really be `dsssl`?  Is it worth changing at the same time?

**Comment by venthur** [comment]
> Shouldn't the `dsl` filetype really be `dsssl`? Is it worth changing at the same time?

I think yes, but I'm no expert on that matter. In any case, I guess that should go into a separate PR then.

**Comment by venthur** [comment]
Hi @brammool is there anything left for me to do in order to get the missing syntax file merged?

Cheers!

**Comment by brammool** [comment]
> Hi @brammool is there anything left for me to do in order to get the
> missing syntax file merged?

It will be included in the next runtime files update.

-- 
Sometimes I think the surest sign that intelligent life exists elsewhere
in the universe is that none of it has tried to contact us.     (Calvin)

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by gquemener** [comment]
It seems that the patch that have been applied on master does not include the [syntax file](runtime/syntax/structurizr.vim) @brammool .

**Comment by dkearns** [comment]
It was included with 89a9c159f23fb7b3e24e6d09068adfc24a73afcb

Runtime files get updated periodically in batches.

**Comment by gquemener** [comment]
My bad, I have found it first time I searched for it. Thanks!

**Pull request closed without merging**